docs/releasing/testing-and-linting.md

```diff
@@ -2,45 +2,80 @@
+
+[Snapshot tests](https://facebook.github.io/jest/docs/en/snapshot-testing.html) are used for preventing unintended changes to our component markup. When the snapshot test runs, it  compares the previously captured snapshot to the current markup.
+
+Visual regression tests help us check for any unintended visual changes to our components. We use [Percy](https://percy.io/) to generate and store screenshots of our components.
+
+## Running linting and tests
+
+### Running all tests locally
 
 To check the whole codebase, run:
 
 ```
 npm test
 ```
 
-This will trigger [standard](https://github.com/standard/standard) and [stylelint](https://github.com/stylelint/stylelint) for linting, and [Jest](https://github.com/facebook/jest) for unit and functional tests.
+This will trigger [standard](https://github.com/standard/standard) and [stylelint](https://github.com/stylelint/stylelint) for linting, and [Jest](https://github.com/facebook/jest) for our testing.
 
 See [Tasks](tasks.md) for details of what `npm test` does.
 
-## SASS linting
+### Running individual tests
+You can run a subset of the test suite that only tests components by running:
+
+    `npm test -- --testPathPattern=src/govuk/components/button`
+
+Note: There's a watch mode that keeps a testing session open waiting for changes that can be used with:
+
+    `npm test -- --watch src/govuk/components/button`
+
+### Running only SASS linting
+
+```
+npm run lint:scss
+```
 
 See [CSS Coding Standards](/docs/contributing/coding-standards/css.md#linting) for details.
 
-## Javascript linting
+### Running only JavaScript linting
+
+```
+npm run lint:js
+```
 
 See [JavaScript Coding Standards](/docs/contributing/coding-standards/js.md#formatting-and-linting) for details.
 
 ## Unit and functional tests with Node.js
 
 We use [Jest](https://jestjs.io/), an automated testing platform with an assertion library, and [Puppeteer](https://pptr.dev/) that is used to control [headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome).
 
-We test individual components and for instance global Sass files. See [all.test.js](../../src/govuk/all.test.js). for examples of global tests we run, and  [checkboxes.test.js](../../src/govuk/components/checkboxes/checkboxes.test.js) for an example of component tests.
+### Component tests
+We write tests for every component to check the output of our Nunjucks code. These are found in `template.test.js` files in each component directory. These Nunjucks tests call `render` on the component examples defined in the component yaml files, and asserts the HTML tags, attributes and classes are as expected. For example: checking that when you pass in an `id` to the component using the Nunjucks macro, it outputs the component with an `id` attribute equal to that value.
```

Comment:
What type of test would component tests be? Unit or functional? I'm wondering if we should sus this out and jot it up as a conscious choice. I _think_ this is functional as we're doing less logic based tests compared to making sure stuff renders properly.

Reply:
I think functional too. I'll add that in here somewhere...